apps/mark-scan/frontend/src/app_root.tsx

```diff
@@ -513,7 +523,13 @@ export function AppRoot({ reload }: Props): JSX.Element | null {
     }
 
     if (pollsState === 'polls_open') {
-      if (isCardlessVoterAuth(authStatus)) {
+      if (
+        isCardlessVoterAuth(authStatus) &&
+        // accepting_paper expects poll worker auth. If the frontend sees accepting_paper but has cardless voter auth,
+        // it means the state hasn't caught up to auth changes. We check that edge case here to avoid flicker ie.
+        // rendering the ballot briefly before rendering the correct "Insert Card" screen
+        stateMachineState !== 'accepting_paper'
+      ) {
```

Comment:
I feel like we've run into these cases before where the backend and frontend have out of sync state. It's pretty confusing. Do you think we could make it more straightforward if the backend was solely responsible for deciding what state we should show and the frontend only rendered screens based on the state machine state (rather than also trying to incorporate auth state)?

Reply:
Yeah I didn't foresee this issue when adding auth polling to the backend. I agree it's pretty awful. In my mind the fix is related to https://github.com/votingworks/vxsuite/issues/3985, which would move the frontend to an exhaustive `switch` to handle all state machine states (and _only_ states, no auth).

cc @arsalansufi who had a question about #3985 - for all states that involve auth, we'll need to patch like this.

Reply:
Thanks for the ping / example! Does make the motivation for https://github.com/votingworks/vxsuite/issues/3985 clear